app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java

```diff
@@ -273,13 +290,49 @@ public void onClick(View v) {
     }
 
     public void goToIndividualChat(View view) {
-        Intent intent = new Intent(ActivityDescriptionActivity.this, ChatActivity.class);
-        String activityChatId = act.getActivityId() + " - chatId";
-        intent.putExtra("ACTIVITY_CHAT_ID", activityChatId);
-        String activityTitle = act.getTitle();
-        intent.putExtra("ACTIVITY_TITLE", activityTitle);
-        startActivity(intent);
+        // Allowing the access to the chat only if the user is registered to the activity
+        if (act.getParticipantId().contains(userId)) {
```

Comment:
Good check, we need it

Reply:
In fact, this is key for the chat of the sport activity. We don't want intruders to join! 😝